lib/rules/shorthand-property-no-redundant-values/index.mjs

```diff
@@ -122,40 +115,44 @@ const rule = (primary, _secondaryOptions, context) => {
 		}
 
 		root.walkDecls((decl) => {
-			if (!isStandardSyntaxDeclaration(decl) || !isStandardSyntaxProperty(decl.prop)) {
-				return;
-			}
+			if (!isStandardSyntaxDeclaration(decl)) return;
 
-			const prop = decl.prop;
-			const value = decl.value;
+			const { prop, value } = decl;
+
+			if (!isStandardSyntaxProperty(prop)) return;
+
+			if (!isStandardSyntaxValue(value)) return;
 
 			const normalizedProp = vendor.unprefixed(prop.toLowerCase());
 
-			if (hasIgnoredCharacters(value) || !isShorthandProperty(normalizedProp)) {
-				return;
-			}
+			if (!isShorthandProperty(normalizedProp)) return;
 
 			/** @type {string[]} */
 			const valuesToShorthand = [];
 
-			valueParser(value).walk((valueNode) => {
-				if (valueNode.type !== 'word') {
-					return;
+			let shouldIgnore = false;
+
+			for (const valueNode of valueParser(value).nodes) {
+				if (isValueDiv(valueNode)) {
+					shouldIgnore = true;
+					break;
+				}
+
+				if (!(isValueWord(valueNode) || isValueFunction(valueNode))) {
+					continue;
 				}
 
 				valuesToShorthand.push(valueParser.stringify(valueNode));
-			});
+			}
+
+			if (shouldIgnore) return;
 
 			if (valuesToShorthand.length <= 1 || valuesToShorthand.length > 4) {
 				return;
 			}
 
-			const shortestForm = canCondense(
-				valuesToShorthand[0] || '',
-				valuesToShorthand[1] || '',
-				valuesToShorthand[2] || '',
-				valuesToShorthand[3] || '',
-			);
+			const [top, right, bottom, left] = valuesToShorthand;
```

Comment:
```suggestion
const [first, second, third, fourth] = valuesToShorthand;
```

e.g. `grid-gap: <'row-gap'> <'column-gap'>`

Reply:
Need to be consistent with canCondense arguments. It is another refactoring chance.